gui/src/app/compileStanProgram/mainJsUrlCache.ts

```diff
@@ -0,0 +1,74 @@
+export const checkMainJsUrlCache = async (
+  stanProgram: string,
+  baseStanWasmServerUrl: string,
+): Promise<string | null> => {
+  const mainJsCache = localStorage.getItem("mainJsCache");
+  if (!mainJsCache) return null;
+  try {
+    const cache = JSON.parse(mainJsCache);
+    const stanProgramHash = stringChecksum(stanProgram);
+    const url = cache[stanProgramHash];
+    if (!url) return null;
+    if (!url.startsWith(baseStanWasmServerUrl)) {
+      // the url is not from the current server
+      return null;
+    }
+    // check to see if the url is still valid
+    const exists = await checkRemoteFileExists(url);
+    if (!exists) {
+      console.warn("mainJsCache url no longer exists");
```

Comment:
Not sure if this is worth warning over -- we expect it to happen when the server restarts, etc

Reply:
removed.